pkg/operators/operators.go

```diff
@@ -71,6 +83,45 @@ type Operator interface {
 	Instantiate(gadgetContext GadgetContext, gadgetInstance any, params *params.Params) (OperatorInstance, error)
 }
 
+type ImageOperator interface {
+	Name() string
+
+	// InstantiateImageOperator will be run to load information about a gadget and also to _possibly_
+	// run the gadget afterward. It should only do things that are required to populate
+	// DataSources and Params. It could use caching to speed things up, if necessary.
+	InstantiateImageOperator(GadgetContext, ocispec.Descriptor) (ImageOperatorInstance, error)
```

Comment:
The plan was to separate the interfaces, so that one operator could be implementing the interfaces of several operator types; e.g. the ebpf image operator could also work as a data operator.